lib/recurly/pricing/index.js

```diff
@@ -170,9 +170,18 @@ Pricing.prototype.plan = function (planCode, meta, done) {
         self.currency(keys(plan.price)[0]);
       }
 
-      debug('set.plan');
-      self.emit('set.plan', plan);
-      resolve(plan);
+      // If we have a coupon, it must be reapplied
+      if (self.items.coupon) {
+        self.coupon(self.items.coupon.code).then(finish, finish);
+      } else {
+        finish();
+      }
```

Comment:
Thus when we change plans, we reapply the potentially plan-dependent coupon.

We don't expose in the API which plans a coupon applies to since that may be sensitive information a merchant doesn't wish to expose publicly. The result is more API calls.